sources/packages/backend/apps/api/src/route-controllers/institution/_tests_/e2e/institution.aest.controller.getInstitutionDetailById.e2e-spec.ts

```diff
@@ -0,0 +1,103 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import {
+  AESTGroups,
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  getAESTToken,
+} from "../../../../testHelpers";
+import {
+  E2EDataSources,
+  createE2EDataSources,
+  createFakeInstitution,
+} from "@sims/test-utils";
+import { InstitutionDetailAPIOutDTO } from "../../models/institution.dto";
+import {
+  INSTITUTION_TYPE_BC_PRIVATE,
+  INSTITUTION_TYPE_BC_PUBLIC,
+} from "@sims/sims-db/constant";
+import { getISODateOnlyString } from "@sims/utilities";
+
+describe("InstitutionAESTController(e2e)-getInstitutionDetailById", () => {
+  let app: INestApplication;
+  let db: E2EDataSources;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+  });
+
+  it("Should return institution details when an institution with given institution id exist.", async () => {
+    // Arrange
+    const institution = await db.institution.save(createFakeInstitution());
+    const mailingAddress = institution.institutionAddress.mailingAddress;
+    const institutionType = await db.institutionType.findOne({
+      select: { id: true, name: true },
+      where: { id: institution.institutionType.id },
+    });
+    const isBCPrivate =
+      INSTITUTION_TYPE_BC_PRIVATE === institution.institutionType.id;
+    const isBCPublic =
+      INSTITUTION_TYPE_BC_PUBLIC === institution.institutionType.id;
+    const expectedInstitutionDetails: InstitutionDetailAPIOutDTO = {
```

Comment:
This is controlled data and we know what they will be in the `expectedInstitutionDetails`. Is there a reason or benefit to have them defined here?

Reply:
Good point